File: server/Models/Ride.cs
using Microsoft.AspNetCore.Identity;

namespace server.Models;

public class Ride
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string UserId { get; set; }
    public Guid? BikeId { get; set; }
    public Guid? StartStationId { get; set; }
    public Guid? EndStationId { get; set; }
    public DateTime? StartedAtUTC { get; set; }
    public DateTime? FinishedAtUTC { get; set; }
    public decimal DistanceMeters { get; set; }
    public decimal FareAmount { get; set; }
    public RideStatus RideStatus { get; set; }

    public IdentityUser? User { get; set; }
    public Bike? Bike { get; set; }
    public Station? StartStation { get; set; }
    public Station? EndStation { get; set; }
}

File: adminClient/interfaces/ride.ts
export enum RideStatus {
  Started = 0,
  Finished = 1,
  Cancelled = 2,
}

export interface Ride {
  id: string;
  userId: string;
  bikeId: string;
  startStationId: string | null;
  endStationId: string | null;
  startedAtUTC: string;
  finishedAtUTC: string | null;
  distanceMeters: number;
  fareAmount: number;
  rideStatus: RideStatus;
}

export interface CreateRideRequest {
  userId: string;
  bikeId: string;
  startStationId: string | null;
  rideStatus: RideStatus;
}

export interface UpdateRideRequest {
  id: string;
  userId: string;
  bikeId: string | null;
  startStationId: string | null;
  endStationId: string | null;
  startedAtUTC: string | null;
  finishedAtUTC: string | null;
  distanceMeters: number;
  fareAmount: number;
  rideStatus: RideStatus;
}
